model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/event/admin/delegate/JpaAdminEventDelegateProvider.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.jpa.event.admin.delegate;
+
+import java.util.UUID;
+
+import javax.persistence.EntityManager;
+
+import org.keycloak.models.map.common.EntityField;
+import org.keycloak.models.map.common.delegate.DelegateProvider;
+import org.keycloak.models.map.events.MapAdminEventEntity;
+import org.keycloak.models.map.events.MapAdminEventEntityFields;
+import org.keycloak.models.map.storage.jpa.JpaDelegateProvider;
+import org.keycloak.models.map.storage.jpa.event.admin.entity.JpaAdminEventEntity;
+
+/**
+ * A {@link DelegateProvider} implementation for {@link JpaAdminEventEntity}.
+ *
+ * @author <a href="mailto:sguilhen@redhat.com">Stefan Guilhen</a>
+ */
+public class JpaAdminEventDelegateProvider extends JpaDelegateProvider<JpaAdminEventEntity> implements DelegateProvider<MapAdminEventEntity> {
+
+    private final EntityManager em;
+
+    public JpaAdminEventDelegateProvider(final JpaAdminEventEntity delegate, final EntityManager em) {
+        super(delegate);
+        this.em = em;
+    }
+
+    @Override
+    public MapAdminEventEntity getDelegate(boolean isRead, Enum<? extends EntityField<MapAdminEventEntity>> field, Object... parameters) {
+        if (getDelegate().isMetadataInitialized()) return getDelegate();
+        if (isRead) {
+            if(field instanceof MapAdminEventEntityFields) {
+                switch ((MapAdminEventEntityFields) field) {
+                    case ID:
+                    case REALM_ID:
+                    case TIMESTAMP:
+                    case EXPIRATION:
+                        return getDelegate();
+
+                    default:
+                        setDelegate(em.find(JpaAdminEventEntity.class, UUID.fromString(getDelegate().getId())));
+                }
+            } else {
+                throw new IllegalStateException("Not a valid client scope field: " + field);
+            }
+        } else {
+            setDelegate(em.find(JpaAdminEventEntity.class, UUID.fromString(getDelegate().getId())));
+        }
+        return getDelegate();
+    }
+}
```

Comment:
@vramik I've added the delegate for now, although we might decide to simply load everything - including the metadata - in read with parameters. You mentioned you would follow this approach for the user sessions, and if you do I'll update this one to be consistent with what you are doing there.

Reply:
Changed, both event entities no longer have a delegate.